.github/workflows/_buildpacks-release.yml

```diff
@@ -177,26 +163,90 @@ jobs:
         with:
           registry: docker.io
           username: ${{ secrets.docker_hub_user }}
           password: ${{ secrets.docker_hub_token }}
 
-      - name: Check if version is already on Docker Hub
-        id: check
-        run: echo "published_to_docker=$(docker manifest inspect "${{ matrix.docker_repository }}:${{ matrix.buildpack_version }}" &> /dev/null && echo 'true' || echo 'false')" >> $GITHUB_OUTPUT
 
-      - name: Tag and publish buildpack
-        if: inputs.dry_run == false && steps.check.outputs.published_to_docker == 'false'
+      - name: Publish to temporary tags
+        env:
+          TARGETS: ${{ toJSON(matrix.targets) }}
+        run: |
+          target_temp_tags=($(jq --exit-status -r "map(.temporary_tag) | join(\" \")" <<< "${TARGETS}"))
+          # Publish each target to a temp tag
+          for i in "${!target_temp_tags[@]}"; do
+            output_dir=$(jq --exit-status -r ".[$i].output_dir" <<< "${TARGETS}")
+            echo "Packaging ${output_dir} into ${target_temp_tags[i]}"
+            pack buildpack package "${target_temp_tags[i]}" --config "${output_dir}/package.toml" -v --publish
+          done
+
+          # If there is more than one target, publish a multi-platform
+          # manifest list / image index to a temp tag.
+          if (( ${#target_temp_tags[@]} > 1 )); then
+            # create a manifest list using platform-specific images created above.
+            docker manifest create "${{ matrix.temporary_tag }}" "${target_temp_tags[@]}"
+
+            # annotate each of the manifest list entries with the correct os/arch
+            for i in "${!target_temp_tags[@]}"; do
+              os=$(jq --exit-status -r ".[$i].os" <<< "${TARGETS}")
+              arch=$(jq --exit-status -r ".[$i].arch" <<< "${TARGETS}")
+              echo "Annotating ${{ matrix.temporary_tag }} / ${target_temp_tags[i]} with ${os}/${arch}"
+              docker manifest annotate "${{ matrix.temporary_tag }}" "${target_temp_tags[i]}" --os "${os}" --arch "${arch}"
+            done
+
+            # Push the manifest list / image index to a temporary tag
+            docker manifest push "${{ matrix.temporary_tag }}"
+          fi
+
+      - name: Promote temporary tags to stable tags
+        if: inputs.dry_run == false
+        env:
+          TARGETS: ${{ toJSON(matrix.targets) }}
+        run: |
+          # Promote target temp tags to stable tags
+          target_temp_tags=($(jq --exit-status -r "map(.temporary_tag) | join(\" \")" <<< "${TARGETS}"))
+          for i in "${!target_temp_tags[@]}"; do
+            perm_tag=$(jq --exit-status -r ".[$i].permanent_tag" <<< "${TARGETS}")
+            crane copy "${target_temp_tags[i]}" "${perm_tag}"
+          done
+          # promote primary image manifest or manifest list to permanent tag
+          crane copy "${{ matrix.temporary_tag }}" "${{ matrix.permanent_tag }}"
+
+      - name: Unpublish temp tags from this run
+        if: always()
```

Comment:
This shouldn't run if `steps.check.outputs.published_to_docker == 'false'` as can be seen here: https://github.com/heroku/buildpacks-nodejs/actions/runs/8712758197/job/23899957917.

Or, if it always runs, it shouldn't error out like it does in the run linked above.

Reply:
> This shouldn't run if `steps.check.outputs.published_to_docker == 'false'`

I think it should run. In both dry run mode and scenarios where we've already published the versioned tag (`heroku/buildpack-procfile:3.0.0`), we still want to make sure the temp tags are deleted (`heroku/buildpack-procfile:_12345`). We never want to leave these around. If any of the tags don't exist (say due to a previous error), the deletion step still exits with a success.

That run errored out due to insufficient scope - which means the docker personal access token doesn't have `delete`. Since we're deleting tags, each repo using this action will need new personal access tokens with `delete` scope. I think that scenario should be an error - it's leaving extra tags around and needs developer intervention.

Reply:
Ah, my bad. I thought it was erroring out because there were no tags to delete. Ignore my earlier comment 😅